### PR TITLE
Add validation against unknown templates and interfaces in transaction filter.

### PR DESCRIPTION
**Pull request by skisel-da**
CHANGELOG_BEGIN
CHANGELOG_END

### CODE REVIEW

File: ledger/participant-integration-api/src/main/scala/platform/index/IndexServiceImpl.scala [hunk 1]
```diff
@@ -103,81 +105,84 @@ private[index] class IndexServiceImpl(
...
-                  .getFlatTransactions(
-                    startExclusive,
-                    endInclusive,
-                    templateFilter,
-                    eventProjectionProperties,
-                  )
-            }
-          },
-          to,
+    withValidatedFilter(transactionFilter, packageMetadataView.current()) {
```
Comment (tudor-da) [inline ledger/participant-integration-api/src/main/scala/platform/index/IndexServiceImpl.scala:108]: Validation at this layer looks a bit off. There is the option of keeping this concern separated and performing this validation in `GrpcTransactionService` in the same manner we do it for the `ledgerEnd` (i.e. expose the packageMetadataView in the `IndexService`), together with the other arguments validation. Wdyt @nmarton-da @skisel-da ?

File: ledger/participant-integration-api/src/main/scala/platform/index/IndexServiceImpl.scala [hunk 2]
```diff
@@ -467,6 +475,67 @@ private[index] class IndexServiceImpl(
...
+    val templatesOrInterfaces = unknownTemplatesOrInterfaces(domainTransactionFilter, metadata)
+
+    if (templatesOrInterfaces.nonEmpty) {
+      Source.failed(
+        invalidArgument(
+          invalidTemplateOrInterfaceMessage(templatesOrInterfaces)
+        )
+      )
+    } else
+      source
```
Comment (tudor-da) [inline ledger/participant-integration-api/src/main/scala/platform/index/IndexServiceImpl.scala:518]: I would either use curlies for both branches or for none 
```suggestion
    } else {
      source
    } 
```

File: ledger/participant-integration-api/src/main/scala/platform/index/IndexServiceImpl.scala [hunk 3]
```diff
@@ -467,6 +475,67 @@ private[index] class IndexServiceImpl(
...
+      unknownInterfaces =
+        inclusiveFilter.interfaceFilters
+          .collect {
+            case interfaceFilter if !metadata.interfaces.contains(interfaceFilter.interfaceId) =>
+              Right(interfaceFilter.interfaceId)
+          }
+      unknownTemplates = inclusiveFilter.templateIds
+        .collect {
+          case templateId if !metadata.templates.contains(templateId) => Left(templateId)
+        }
```
[the remark this answers is not in the document]
Comment (tudor-da) [reply ledger/participant-integration-api/src/main/scala/platform/index/IndexServiceImpl.scala:496]: might read better

**Review by tudor-da — APPROVED** [review]

File: ledger/participant-integration-api/src/main/scala/platform/index/IndexServiceImpl.scala (hunk 1 above)
Comment (skisel-da) [reply ledger/participant-integration-api/src/main/scala/platform/index/IndexServiceImpl.scala:108]: An attempt I made is here:
https://github.com/digital-asset/daml/pull/14737

**Pull request merged**